### PR TITLE
[website]Support multi version for swagger

### PR DESCRIPTION
**Pull request by tuteng**
Fixes https://github.com/apache/pulsar/issues/3505

Master Issue: https://github.com/apache/pulsar/issues/3505

### Motivation

Currently, there is no version management in rest API, and the latest version will be returned every time. This is usually confusing, so add version management to rest API. The new URL is similar to this:  
```
http://localhost:3000/functions-rest-api?version=2.4.0
http://localhost:3000/functions-rest-api?version=2.3.0
http://localhost:3000/functions-rest-api?version=master
```

### Modifications

* Add version parse
* Add version for link

### Verifying this change

image

When content is published, the folder below needs to be added according to the version number above.

image

My local test did not generate JSON files before version 2.3, but it needs to be added when it is released.

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by jiazhai** [comment]
run cpp tests

**Comment by sijie** [comment]
run cpp tests

**Comment by jiazhai** [comment]
run java8 tests

File: src/gen-swagger.sh [hunk 1]
```diff
@@ -22,10 +22,43 @@ PULSAR_PATH=$(git rev-parse --show-toplevel)
 
 cd $PULSAR_PATH
 
-echo "Generating swagger json file ..."
+git checkout v2.3.0
```
[the remark this answers is not in the document]
Comment (tuteng) [reply src/gen-swagger.sh]: @merlimat PTAL

**Comment by tuteng** [comment]
run integration tests

**Comment by tuteng** [comment]
run integration tests

**Pull request merged**